### PR TITLE
Add discussion for template and conclusion

### PR DESCRIPTION
**Pull request by weixuanfu**
Relate issue #4 and #5

### CODE REVIEW

File: content/05.discussion.md [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+TPOT-DS is able to identify the most meaningful group of features to include in the prediction pipeline.
 We assessed TPOT-DS's out-of-sample prediction accuracy compared to standard TPOT and XGBoost, another state-of-the-art machine learning method.
 We applied TPOT-DS to real-world expression data to demonstrate the identification of biologically relevant groups of genes.
 
-Implemented with a strongly typed GP, Template allows users to pre-specify a particular pipeline structure, which speeds up the automation computation time and provides potentially more interpretable results.
-Consequently, Template enables the comparison between the two TPOT implementations, with and without DS.
+Implemented with a strongly typed GP, Template provides more flexibility by allowing users to pre-specify a particular pipeline structure based on their knowledge, which speeds up AutoML process and provides potentially more interpretable results.
+For example, for high-dimensional data, dimension reduction algorithms or feature selection algorithms are preferred to add into pipeline via Template for reducing computation time and potentially detecting important features.
```
Comment (trangdata) [inline content/05.discussion.md]: For example, in high-dimensional data, dimensionality reduction or feature selection algorithms are preferably included at the beginning of the pipelines via Template to  identify important features and, meanwhile, reduce computation time.

File: content/05.discussion.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+For dataset mixed with categorical features, preprocessing operators for encoding those features, like one-hot encoder, should be specified in pipeline structure for improving pipelines' performance. In this study, the Dataset Selector was specified as the first step of pipeline via Template, which enables the comparison between the two TPOT implementations, with and without DS.
```
Comment (trangdata) [inline content/05.discussion.md]: For datasets with categorical features, preprocessing operators for encoding those features, like one-hot encoder, should be specified in the pipeline structure to improve pipelines' performance.
Template was utilized in this study to specify the Dataset Selector as the first step of the pipeline, which enables the comparison between the two TPOT implementations, with and without DS.

**Review by trangdata — COMMENTED** [review]
Thank you for the changes! I made a few minor tweaks here!

File: content/05.discussion.md [hunk 3]
```diff
@@ -37,13 +40,15 @@ Additionally, complexity reduction results in more efficient computation, which
...
 A future design to support tree structures for Template will enable TPOT-DS to identify more than one subset that have high predictive power of the outcome.
 A new operator that combines the data subsets will prove useful in this design.
 Extensions of TPOT-DS will also involve overlapping subsets, which will require pipeline complexity reformulation beyond the total number of operators included in a pipeline.
 Specifically, in the case of overlapping subsets, the number of features in the selected subset(s) is expected to be an element of the complexity calculation.
 Extension of TPOT-DS to GWAS is straightforward.
 However, because of the low predictive power of variants in current GWAS, alternative metrics beside accuracy, balanced accuracy or area under the receiving operator characteristic curve will need to be designed and included in the fitness function of TPOT's evolutionary algorithm.
 
-[Conclusion]
\ No newline at end of file
+In this study, we added DS and Template into TPOT to enhance its performance on high-dimensional data by reducing computation time and simplifying pipeline structure.
```
Comment (trangdata) [inline content/05.discussion.md]: In this study, we developed two new operators for TPOT, Dataset Selector and Template, to enhance its performance on high-dimensional data by simplifying pipeline structure and reducing the computational expense.

File: content/05.discussion.md [hunk 4]
```diff
@@ -37,13 +40,15 @@ Additionally, complexity reduction results in more efficient computation, which
...
+With domain knowledge, DS can potentially help users to narrow down important features for further interpretation.
```
Comment (trangdata) [inline content/05.discussion.md]: With domain knowledge, Dataset Selector can help users narrow down important features for further interpretation.

File: content/05.discussion.md [hunk 5]
```diff
@@ -37,13 +40,15 @@ Additionally, complexity reduction results in more efficient computation, which
...
+Template largely increase flexibility of using TPOT via customizing pipeline structure, which potentially enrich the application of AutoML on different real world problems.
```
Comment (trangdata) [inline content/05.discussion.md:53]: I like the use of 'customizing' here.

**Comment by weixuanfu** [comment]
Thank you for reviews. I just pushed a commit to revise those sentences.

**Review by trangdata — APPROVED** [review]
Great! Thanks a lot!

**Comment by trangdata** [comment]
closes #4 and #5

**Pull request merged**